src/bdf_utils.jl

```diff
@@ -23,3 +23,116 @@ function R!(k, ρ, cache)
     end
   end
 end
+
+# This functions takes help of D2 array to create backward differences array D
+# Ith row of D2 keeps Ith order backward differences (∇ⁱyₙ)
+function backward_diff!(D, D2, k, flag=true)
+  if flag
+    D[1] = D2[1,1]
+  end
+  for i = 2:k
+    for j = 1:(k-i+1)
+      D2[i,j] = D2[i-1,j] - D2[i-1,j+1]
+    end
+    if flag
+      D[i] = D2[i,1]
+    end
+  end
+end
+
+global const γₖ = [1//1, 3//2, 11//6, 25//12, 137//60, 49//20]
+
+function stepsize_and_order!(cache, est, estₖ₋₁, estₖ₊₁, h, k)
+  zₛ = 1.2
+  zᵤ = 0.1
+  Fᵤ = 10
+
+  expo = 1/(k+1)
+  z = zₛ * ((est)^expo)
+  F = inv(z)
+
+  hₖ₋₁ = 0.0
+  hₖ₊₁ = 0.0
+
+  if z <= zₛ
+    # step is successful
+    # precalculations
+    if z <= zᵤ
+      hₖ = Fᵤ * h
+    elseif zᵤ < z <= zₛ
+      hₖ = F * h
+    end
+
+    if k > 1
+      expo = 1/k
+      zₖ₋₁ = 1.3 * ((estₖ₋₁)^expo)
+      Fₖ₋₁ = inv(zₖ₋₁)
+      if zₖ₋₁ <= 0.1
+        hₖ₋₁ = 10 * h
+      elseif 0.1 < zₖ₋₁ <= 1.3
+        hₖ₋₁ = Fₖ₋₁ * h
+      end
+    end
+
+    expo = 1/(k+2)
+    zₖ₊₁ = 1.4 * ((estₖ₊₁)^expo)
+    Fₖ₊₁ = inv(zₖ₊₁)
+
+    if zₖ₊₁<= 0.1
```

Comment:
here is an issue, when `zₖ₊₁` is outside of intervals, `h_k+1` will be assigned to zero then condition at [line no 94](https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/450/files#diff-2deacee6c498fd687442bd4fc313fe45R94) will not get satisfied and order will not increase.
for `ode_linear` prob, after some initial steps, order is not increasing due to this issue.
```
(k, t, dt) = (1, 0.0, 9.82256641505735e-6)
(k, t, dt) = (1, 9.82256641505735e-6, 8.840309773551615e-6)
(k, t, dt) = (2, 1.8662876188608963e-5, 8.840309773551615e-5)
(k, t, dt) = (2, 0.00010706597392412511, 0.0008840309773551614)
zₖ₊₁ = 0.6308668055739214
(hₖ₊₁, hₙ) = (0.0014012957561635024, 0.008840309773551614)
(k, t, dt) = (2, 0.0009910969512792866, 0.008840309773551614)
zₖ₊₁ = 1.1205160131791863
(hₖ₊₁, hₙ) = (0.007889498828731085, 0.08840309773551615)
(k, t, dt) = (1, 0.009831406724830901, 0.08840309773551615)
(k, t, dt) = (1, 0.009831406724830901, 0.05007901218339588)
zₖ₊₁ = 1.719673582558603
(hₖ₊₁, hₙ) = (0.0, 0.049196025773848386)
(k, t, dt) = (1, 0.05991041890822678, 0.05007901218339588)
zₖ₊₁ = 1.9044260419789238
(hₖ₊₁, hₙ) = (0.0, 0.04727056107732615)
(k, t, dt) = (1, 0.10998943109162265, 0.05007901218339588)
zₖ₊₁ = 1.903623686593515
(hₖ₊₁, hₙ) = (0.0, 0.04700409617773053)
(k, t, dt) = (1, 0.16006844327501854, 0.05007901218339588)
zₖ₊₁ = 1.903551086995541
(hₖ₊₁, hₙ) = (0.0, 0.04696357785130702)
(k, t, dt) = (1, 0.21014745545841443, 0.05007901218339588)
zₖ₊₁ = 1.903583656012059
(hₖ₊₁, hₙ) = (0.0, 0.04695602368130636)
(k, t, dt) = (1, 0.2602264676418103, 0.05007901218339588)
zₖ₊₁ = 1.903629538310494
(hₖ₊₁, hₙ) = (0.0, 0.046953392075560194)
(k, t, dt) = (1, 0.3103054798252062, 0.05007901218339588)
zₖ₊₁ = 1.9036753242727178
(hₖ₊₁, hₙ) = (0.0, 0.04695156072777662)
(k, t, dt) = (1, 0.36038449200860206, 0.05007901218339588)
zₖ₊₁ = 1.9037191433129206
(hₖ₊₁, hₙ) = (0.0, 0.04694991944944634)
(k, t, dt) = (1, 0.41046350419199795, 0.05007901218339588)
zₖ₊₁ = 1.903760817702772
(hₖ₊₁, hₙ) = (0.0, 0.04694837484280412)
(k, t, dt) = (1, 0.46054251637539384, 0.05007901218339588)
zₖ₊₁ = 1.9038004139589966
(hₖ₊₁, hₙ) = (0.0, 0.0469469097266692)
(k, t, dt) = (1, 0.5106215285587897, 0.05007901218339588)
zₖ₊₁ = 1.9038380299239428
(hₖ₊₁, hₙ) = (0.0, 0.04694551830604183)
(k, t, dt) = (1, 0.5607005407421856, 0.05007901218339588)
zₖ₊₁ = 1.9038737636634548
(hₖ₊₁, hₙ) = (0.0, 0.04694419662460158)
(k, t, dt) = (1, 0.6107795529255814, 0.05007901218339588)
zₖ₊₁ = 1.9039077091075043
(hₖ₊₁, hₙ) = (0.0, 0.04694294115177142)
(k, t, dt) = (1, 0.6608585651089772, 0.05007901218339588)
zₖ₊₁ = 1.9039399556152738
(hₖ₊₁, hₙ) = (0.0, 0.04694174856707545)
(k, t, dt) = (1, 0.7109375772923731, 0.05007901218339588)
zₖ₊₁ = 1.90397058810759
(hₖ₊₁, hₙ) = (0.0, 0.046940615721051175)
(k, t, dt) = (1, 0.7610165894757689, 0.05007901218339588)
zₖ₊₁ = 1.903999687273196
(hₖ₊₁, hₙ) = (0.0, 0.04693953962250189)
(k, t, dt) = (1, 0.8110956016591647, 0.05007901218339588)
zₖ₊₁ = 1.9040273297767105
(hₖ₊₁, hₙ) = (0.0, 0.04693851742996507)
(k, t, dt) = (1, 0.8611746138425606, 0.05007901218339588)
zₖ₊₁ = 1.9040535884581693
(hₖ₊₁, hₙ) = (0.0, 0.046937546444168476)
(k, t, dt) = (1, 0.9112536260259564, 0.05007901218339588)
zₖ₊₁ = 1.9040785325228897
(hₖ₊₁, hₙ) = (0.0, 0.046936624100886955)
(k, t, dt) = (1, 0.9613326382093522, 0.03866736179064778)
zₖ₊₁ = 2.0657718367441675
(hₖ₊₁, hₙ) = (0.0, 0.04585789653324142)

```

Reply:
The linear ODE is going unstable (all the time) so it should prefer lower orders in this algorithm. See what happens with something like the Robertson equation.

Reply:
Okay